packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx

```diff
@@ -80,6 +129,30 @@ export const TransactionReviewModalContent = ({
     );
 
     const isTradingAction = !!precomposedForm?.isTrading;
+
+    const deadline = timestamp + SOLANA_TX_TIMEOUT_MS;
+
+    // check if transaction is still valid
+    useEffect(() => {
+        if (isTradingAction) {
```

Comment:
Shouldn't this depend on showSolanaTimer? This would now run for all networks, unless I'm missing something?